library/std/src/os/unix/net/ancillary.rs

```diff
@@ -78,6 +78,36 @@ pub(super) fn send_vectored_with_ancillary_to(
     }
 }
 
+/// The size in octets of a control message containing `n` `T`s.
+///
+/// # Panics
+///
+/// If the size is so large that it causes arithmetic overflow.
+fn ancillary_size_data<T>(n: usize) -> u32 {
+    let r = u32::try_from(n.checked_mul(size_of::<T>()).unwrap()).unwrap();
+    unsafe { libc::CMSG_SPACE(r) }
+}
+
+/// The size in octets of a control message containing `n` file descriptors.
+///
+/// # Panics
+///
+/// If the size is so large that it causes arithmetic overflow.
+#[unstable(feature = "unix_socket_ancillary_data", issue = "76915")]
+pub fn ancillary_size_fds(n: usize) -> u32 {
+    ancillary_size_data::<RawFd>(n)
+}
```

Comment:
The type-specific methods don't seem particularly useful, they only avoid having to type the turbofish. Imo the uses be covered by adding examples to `ancillary_size_data`, making it public and removing the other methods.

`SocketCred` and `RawFd` aren't private types, so there's no reason to hide them.

Well, the one downside I can see is that it would tie the rust struct size to the underlying libc type needed for cmsg. But they pretty much serve the purpose of being thin wrappers for the platform types anyway, so that doesn't seem like a real loss.